javaagent/sdk-config.yaml

```diff
@@ -36,3 +38,346 @@ meter_provider:
+#
+# For example, the property "otel.instrumentation.common.default-enabled" is resolved by reading: .instrumentation.java.common.default-enabled
+#
+# Some system properties / env vars cannot be configured via declarative configuration, normally because they are resolved in the application lifecycle before declarative configuration is available. The set of unsupported properties includes:
+# - otel.javaagent.enabled
+# - otel.javaagent.configuration-file
+# - otel.javaagent.extensions
+# - otel.javaagent.logging
+# - otel.resource.providers.[provider].enabled
+# TODO: add automation to keep this in sync with system property / env var documentation: https://opentelemetry.io/docs/zero-code/java/agent/
+instrumentation:
+  java:
+    common:
+      default-enabled: true
+      db-statement-sanitizer:
+        enabled: true
+      enduser:
+        id:
+          enabled: false
+        role:
+          enabled: false
+        scope:
+          enabled: false
+      peer-service-mapping: {}
+      experimental:
+        controller-telemetry:
+          enabled: true
+        span-suppression-strategy: semconv
+    http:
+      client:
+        capture-request-headers: []
+        capture-response-headers: []
+        emit-experimental-telemetry: false
+      server:
+        capture-request-headers: []
+        capture-response-headers: []
+        emit-experimental-telemetry: false
+      known-methods:
+        - CONNECT
+        - DELETE
+        - GET
+        - HEAD
+        - OPTIONS
+        - PATCH
+        - POST
+        - PUT
+        - TRACE
+    messaging:
+      experimental:
+        receive-telemetry:
+          enabled: false
+    methods:
+      enabled: true
+      include: ""
+    external-annotations:
+      enabled: true
+    akka-actor:
+      enabled: true
+    akka-http:
+      enabled: true
+    axis2:
+      enabled: true
+    camel:
+      enabled: true
+    cassandra:
+      enabled: true
+    cxf:
+      enabled: true
+    apache-dbcp:
+      enabled: true
+    apache-dubbo:
+      enabled: true
+    geode:
+      enabled: true
+    apache-httpasyncclient:
+      enabled: true
+    apache-httpclient:
+      enabled: true
+    kafka:
+      enabled: true
+    jsf-myfaces:
+      enabled: true
+    pekko-actor:
+      enabled: true
+    pekko-http:
+      enabled: true
+    pulsar:
+      enabled: true
+    rocketmq-client:
+      enabled: true
+    apache-shenyu:
+      enabled: true
+    struts:
+      enabled: true
+    tapestry:
+      enabled: true
+    tomcat:
+      enabled: true
+    wicket:
+      enabled: true
+    armeria:
+      enabled: true
+    async-http-client:
+      enabled: true
+    aws-lambda:
+      enabled: true
+    aws-sdk:
+      enabled: true
+    azure-core:
+      enabled: true
+    clickhouse:
+      enabled: true
+    couchbase:
+      enabled: true
+    c3p0:
+      enabled: true
+    dropwizard-views:
+      enabled: true
+    dropwizard-metrics:
+      enabled: true
```

Comment:
This instrumentation is not enbabled by default....do we want it enabled here? This also applies to:

* jdbc-datasource
* micrometer
* mybatis
* spring-batch

Reply:
it should reflect the default values I think

Reply:
I had a change of heart on this. I realized that it would be near a huge effort to go and enumerate all the config options for all the instrumentation, especially since many are buried in the code without documentation. It would also be fragile, causing the config to break when breaking changes happen to the config, eroding user confidence.

Instead, I opted to give a thorough explanation of the mapping rules from system properties to declarative config YAML, and demonstrate the configuration of just a couple of modules as an example, leaving the user to apply the rules to configure additional instrumentation modules.

I also opted to base the sample config file on [sdk-migration-config.yaml](https://github.com/open-telemetry/opentelemetry-configuration/blob/v0.3.0/examples/sdk-migration-config.yaml), which is much more exhaustive and includes helpful explanatory comments.